File: NetBeansProjects/java/Payment/src/main/java/com/maicol/payment/Person.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.payment;

/**
 *
 * @author Maicol
 */

   public abstract class Person {
    private final String name;
    private final int salary;
    private final int overtime;
    private final String scheduleType;

    public Person(String name, int salary, int overtime, String scheduleType) {
        this.name = name;
        this.salary = salary;
        this.overtime = overtime;
        this.scheduleType = scheduleType;
    }

    public String getName() {
        return name;
    }

    public int getSalary() {
        return salary;
    }

    public int getOvertime() {
        return overtime;
    }

    public String getScheduleType() {
        return scheduleType;
    }

    public abstract double calculateSalary();
}

File: php/exercises/1.payment/php/person.php
<?php

abstract class person{  
    private string $name;
    private int $salary;
    private int $overtime;
    private string $scheduleType;

    function __construct(string $name, int $salary, int $overtime, string $scheduleType){
        $this->name=$name;
        $this->salary=$salary;
        $this->overtime=$overtime;
        $this->scheduleType=$scheduleType;
    }
    public function getName():string{
        return $this->name;
    }
    public function getSalary():int{
        return $this->salary;
    }
    public function getOvertime():int{
        return $this->overtime;
    }
    public function getScheduleType():string{
        return $this->scheduleType;
    }
    public abstract function calculateSalary():float;
}
?>